packages/docregistry/src/context.ts

```diff
@@ -362,13 +362,34 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
     }
     let oldPath = change.oldValue && change.oldValue.path;
     let newPath = change.newValue && change.newValue.path;
-    if (newPath && oldPath === this._path) {
-      this.session.setPath(newPath);
-      const localPath = this._manager.contents.localPath(newPath);
-      this.session.setName(PathExt.basename(localPath));
-      this._path = newPath;
-      this._updateContentsModel(change.newValue as Contents.IModel);
-      this._pathChanged.emit(this._path);
+
+    if (newPath) {
+      let changeModel = change.newValue
+      // When folder name changed, `oldPath` is `foo`, `newPath` is `bar` and `this._path` is `foo/test`,
+      // we should update `foo/test` to `bar/test` as well
+      if (oldPath !== this._path && this._path.indexOf(oldPath) == 0) {
+        newPath = this.path.replace(new RegExp(`^${oldPath}`), newPath);
+        oldPath = this._path;
+        // Update client file model
+        changeModel = {
+          last_modified: change.newValue.created,
```

Comment:
If we don't update last_modified，It will break when you save the file, since the disk last_modified time was already changed (folder moved), but client last_modified time was previous, so it will be conflict.